### PR TITLE
Metadata update script

### PR DESCRIPTION
**Pull request by betaboon**
depends on #116 

**motivation:**
i noticed that some kernel-packages and patches weren't up-to-date.

**description:**
i adopted the `metadata.nix` approach.
- `metadata.nix` contains versions and hashes for all packages.
- all packages use the information from `metadata.nix`
- `metadata.nix` can be automatically updated using `update.sh`

### CODE REVIEW

**Comment by henrytill** [comment]
Diff against #116:
https://github.com/betaboon/musnix/compare/overlay-2020...betaboon:metadata-update-script

(leaving this here for my own convenience)

**Comment by henrytill** [comment]
Looks like this branch needs another rebase.

**Comment by betaboon** [comment]
@henrytill done.

i think this could be refactored even more (seeing the redundancy in the code now) but maybe that should be done in a next step :)

File: pkgs/os-specific/linux/kernel/update.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i bash -p nixfmt
+
```
Comment (henrytill) [inline pkgs/os-specific/linux/kernel/update.sh:3]: Could we add:

```bash
set -o errexit
set -o pipefail
```

**Comment by henrytill** [comment]
> i think this could be refactored even more (seeing the redundancy in the code now) but maybe that should be done in a next step :)

Understood.  I have mixed feelings about maintaining Bash scripts, but this is great for an initial pass.  We can certainly iterate on it from here. ...

File: pkgs/os-specific/linux/kernel/update.sh (hunk 1 above)
Comment (henrytill) [reply pkgs/os-specific/linux/kernel/update.sh:3]: I went ahead this and added in 4213b35.

**Comment by henrytill** [comment]
Thanks again, @betaboon

**Pull request merged**